static/js/admin/src/AbstractForm.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Implements functionality and UI for the generic add container form
+ */
+define('AbstractForm', [ 'jquery', 'jquery-ui', 'underscore', 'RemoteStateChangeMonitor', 'ModalCreate',
+		'ModalLoadingOverlay', 'ResultObject', 'AlertHandler'],
+	function($, ui, _, RemoteStateChangeMonitor, ModalCreate, ModalLoadingOverlay, ResultObject) {
+		function AbstractForm(options) {}
+
+		AbstractForm.prototype.open = function(resultObject) {
+			var dialogBox = new ModalCreate(this.options);
+			var formContents = dialogBox.formContents(resultObject);
+			var self = this;
+			this.closed = false;
+
+			this.dialog = $("<div class='containingDialog'>" + formContents + "</div>");
+			this.$form = this.dialog.first();
+			this.dialog.dialog = dialogBox.modalDialog(this.dialog, self);
+
+			if (this.containerType) {
```

Comment:
This seems very specific to the create container form, so I am wondering if there would be a way to abstract this